src/mattack_actors.cpp

```diff
@@ -291,8 +291,26 @@ mattack_actor *gun_actor::clone() const
     return new gun_actor( *this );
 }
 
+// @todo Enforce load order (add finalization?) and allow calculating this at json load
+int gun_actor::get_range() const
+{
+    if( target_range > 0 ) {
+        return target_range;
+    }
+
+    item gun( gun_type );
+    itype_id ammo = ( ammo_type != "NULL" ) ? ammo_type : gun.ammo_default();
+
+    if( ammo != "NULL" ) {
```

Comment:
You need to check for `null` not `NULL` ammo. The entire check is redundant in any case as for guns it is no an error to call `ammo_set` with null ammo.